test/test.jl

```diff
@@ -68,82 +64,73 @@ end
     @test all(A[1:8] .== 1:8)
     @test all(A[9:16] .== 1:8)
 
     A = ArrayT{CartesianIndex{2}}(undef, 16, 16)
-    wait(index_cartesian_global(backend, 8)(A, ndrange=size(A)))
+    wait(index_cartesian_global(backend(), 8)(A, ndrange=size(A)))
     @test all(A .== CartesianIndices(A))
 
     A = ArrayT{CartesianIndex{1}}(undef, 16, 16)
-    wait(index_cartesian_global(backend, 8)(A, ndrange=length(A)))
+    wait(index_cartesian_global(backend(), 8)(A, ndrange=length(A)))
     @test all(A[:] .== CartesianIndices((length(A),)))
 
     # Non-multiplies of the workgroupsize
     A = ArrayT{Int}(undef, 7, 7)
-    wait(index_linear_global(backend, 8)(A, ndrange=length(A)))
+    wait(index_linear_global(backend(), 8)(A, ndrange=length(A)))
     @test all(A .== LinearIndices(A))
 
     A = ArrayT{Int}(undef, 5)
-    wait(index_linear_local(backend, 8)(A, ndrange=length(A)))
+    wait(index_linear_local(backend(), 8)(A, ndrange=length(A)))
     @test all(A .== 1:5)
 end
 
-@testset "indextest" begin
-    indextest(CPU(), Array)
-    if has_cuda_gpu()
-        indextest(CUDADevice(), CuArray)
-    end
-end
-
 @kernel function constarg(A, @Const(B))
     I = @index(Global)
     @inbounds A[I] = B[I]
 end
 
 @testset "Const" begin
-    let kernel = constarg(CPU(), 8, (1024,))
+    let kernel = constarg(backend(), 8, (1024,))
         # this is poking at internals
         iterspace = NDRange{1, StaticSize{(128,)}, StaticSize{(8,)}}();
-        ctx = KernelAbstractions.mkcontext(kernel, 1, nothing, iterspace, Val(NoDynamicCheck()))
-        AT = Array{Float32, 2}
+        ctx = if backend == CPU
+            KernelAbstractions.mkcontext(kernel, 1, nothing, iterspace, Val(NoDynamicCheck()))
+        else
+            KernelAbstractions.mkcontext(kernel, nothing, iterspace)
+        end
+        AT = if backend == CPU
+            Array{Float32, 2}
+        else
+            DeviceArrayT{Float32, 2, 1} # AS 1
+        end
         IR = sprint() do io
             code_llvm(io, KernelAbstractions.Cassette.overdub,
                      (typeof(ctx), typeof(kernel.f), AT, AT),
                      optimize=false, raw=true)
         end
-        @test occursin("!alias.scope", IR)
-        @test occursin("!noalias", IR)
-    end
-
-    if has_cuda_gpu()
-        let kernel = constarg(CUDADevice(), 8, (1024,))
-            # this is poking at internals
-            iterspace = NDRange{1, StaticSize{(128,)}, StaticSize{(8,)}}();
-            ctx = KernelAbstractions.mkcontext(kernel, nothing, iterspace)
-            AT = CUDA.CuDeviceArray{Float32, 2, CUDA.AS.Global}
-            IR = sprint() do io
-                CUDA.code_llvm(io, KernelAbstractions.Cassette.overdub,
-                               (typeof(ctx), typeof(kernel.f), AT, AT),
-                               kernel=true, optimize=true)
-            end
+        if backend == CPU
+            @test occursin("!alias.scope", IR)
+            @test occursin("!noalias", IR)
+        else
```

Comment:
```suggestion
        elseif backend == CUDA
```

Reply:
Yeah I have something similar locally. We can't directly compare the backend (since `CUDADevice` might not exist when `ROCKernels is tested in the future), so I currently just check the relevant env. var.